### ISSUE TITLE
"Stage All" Icon is disabled

### ISSUE THREAD

**Issue by ElfSundae** [ForkIssues/Tracker#784]
> Fork 1.0.85.2 , macOS 10.15

If I remember correctly, in previous versions the "Stage All" icon can always be clicked if there are uncommitted files. But in recent versions, its behavior seems to have changed. Is it?
![image](https://user-images.githubusercontent.com/526008/67280387-9f6bf100-f4ff-11e9-837f-97a7d94d7579.png)

For details in the current version:

- If no files are selected (highlighted), the "Stage All" / "Unstage All" **icon** can work, but the "Stage All" / "Unstage All" **button** (press Option key) is disabled:
![image](https://user-images.githubusercontent.com/526008/67282258-ee1b8a00-f503-11e9-86ad-1b31a1a37b88.png)
![image](https://user-images.githubusercontent.com/526008/67282362-3044cb80-f504-11e9-9b52-664f0cc42cd8.png)

   Expectation: the button can also work.

- If there are files in both "Unstaged Changes" window and "Staged Changes" window, the "Stage All" / "Unstage All" icon and button are disabled:
![image](https://user-images.githubusercontent.com/526008/67283176-bdd4eb00-f505-11e9-842a-c63888792fe5.png)

**Comment by ElfSundae** [comment]
In my opinion, if there are uncommitted files, it should be:

```
icon.enabled = true

icon.func = unstagedChanges.files.count ? stageAll() : unstageAll()

stageButton.enabled = unstagedChanges.selected.count
stageAllButton.enabled = unstagedChanges.files.count

unstageButton.enabled = stagedChanges.selected.count
unstageAllButton.enabled = stagedChanges.files.count
```

**Comment by Finesse** [comment]
Meanwhile the keyboard shortcut (⌥⇧⌘S) to stage/unstage all works, even when some files are staged and some are unstaged.

P.S. I have this issue too

**Comment by DanPristupov** [comment]
Fixed in 1.0.86. Thank you for the bug report.
